aggregators/codec.go

```diff
@@ -122,6 +122,7 @@ func (k *serviceAggregationKey) ToProto() *aggregationpb.ServiceAggregationKey {
 	pb.ServiceEnvironment = k.ServiceEnvironment
 	pb.ServiceLanguageName = k.ServiceLanguageName
 	pb.AgentName = k.AgentName
+	pb.GlobalLabelsStr = []byte(k.GlobalLabelsStr)
```

Comment:
I wonder if this and the below `k.GlobalLabelsStr = string(pb.GlobalLabelsStr)` is continuously creating copies and we should store everything as strings or find an alternative.

Either way, this is not changing the previous behaviour so approving!

Reply:
It has to be a `[]byte` because the contents may not be valid UTF-8, which is required for protobuf strings.

Reply:
> https://protobuf.dev/programming-guides/proto3/#scalar

String must always contain UTF-8 as per proto requirements. Since we encode the global labels as proto I don't think it will hold true.

Reply:
Oops, missed axw's message 😅